### PR TITLE
commit progress organisation metrics #3146

### PR DESCRIPTION
**Pull request by salomon-j**

### CODE REVIEW

File: grails-app/controllers/au/org/ala/ecodata/OrganisationController.groovy [hunk 1]
```diff
@@ -114,4 +114,44 @@ class OrganisationController {
         values
     }
 
+    def organisationMetrics(String id) {
+
+        def organisation = Organisation.findByOrganisationId(id)
+
+        boolean approvedOnly = true
+        boolean targetsOnly = false
```
Comment (chrisala) [inline grails-app/controllers/au/org/ala/ecodata/OrganisationController.groovy:122]: Organisations don't have a concept of targets so we should delete targetsOnly and includeTargets

File: grails-app/services/au/org/ala/ecodata/OrganisationService.groovy [hunk 2]
```diff
@@ -217,4 +219,65 @@ class OrganisationService {
...
+
+    /**
+     * Returns the reportable metrics for a organisation as determined by the organisation output targets and activities
+     * that have been undertaken.
+     * @param id identifies the organisation.
+     * @return a Map containing the aggregated results.
+     *
+     */
+    def organisationMetrics(String id, targetsOnly = false, approvedOnly = false, List scoreIds = null, Map aggregationConfig = null, boolean includeTargets = true) {
+        def org = Organisation.findByOrganisationId(id)
```
Comment (chrisala) [inline grails-app/services/au/org/ala/ecodata/OrganisationService.groovy:231]: Organisations don't have a concept of targets so we should delete targetsOnly and includeTargets

File: grails-app/services/au/org/ala/ecodata/OrganisationService.groovy [hunk 3]
```diff
@@ -217,4 +219,65 @@ class OrganisationService {
...
+            } else if (scoreIds) {
+                toAggregate = Score.findAllByScoreIdInList(scoreIds)
+            } else {
+                toAggregate = targetsOnly ? Score.findAllByIsOutputTarget(true) : Score.findAll()
+            }
+
+            List outputSummary = reportService.organisationSummary(id, toAggregate, approvedOnly, aggregationConfig) ?: []
+
+            // Add project output target information where it exists.
+            if (includeTargets) {
```
Comment (chrisala) [inline grails-app/services/au/org/ala/ecodata/OrganisationService.groovy:247]: Delete this section as organisations cannot have targets

**Review by chrisala — CHANGES_REQUESTED** [review]

**Comment by chrisala** [comment]
@salomon-j this got moved to a different branch didn't it?  It is safe to close this?

**Comment by salomon-j** [comment]
> @salomon-j this got moved to a different branch didn't it? It is safe to close this?

yes it is safe to close it, I've created a new branch that was merged into master.

**Pull request closed without merging**